TDDWithURMAndKotlin/part3_jump.md

```diff
@@ -0,0 +1,176 @@
+- Needs to easily change the pointer of execution to any other instruction
+
+We could need more functionality from the `instructionSet` but having this 3 requirements tell us we need a class and will let us at least write the minimum necessary to fullfill our failing test. First we need to change the test adding a type for `instructionSet`:
+
+```kotlin
+internal class OperationsTest {
+    //…
+
+    val instructionSet = InstructionSet()
+}
+```
+
+And we can use it in the test, but wait, the test requires the `instructionSet` to have a `current` property. Let's create the class and add the property to remove the _compile-time error_:
+
+```kotlin
+class InstructionSet {
+    var current: Int = 0
+}
+```
+
+We use `var` because we will be setting the value to a different number on every execution of the instruction set. This class just holds data and actually doesn't need to do anything else for our current porpouse. Our test should pass now right? No compile time errors and it's quite straight forward.
+
+Well, not exactly. If we run the classic: `gradle check` we will get:
+
+```none
+expected: <1> but was: <0>
+org.opentest4j.AssertionFailedError: expected: <1> but was: <0>
+...
+```
+
+Hmmm we forgot to add an implementation to the function! Duh! Well… time to write the actual code:
+
+```kotlin
+fun jump(registry: Registry, positionX: Int, positionY: Int, instructionSet: InstructionSet, instruction: Int) {
+    instructionSet.current = instruction
+}
+```
+
+Easy peasy, now the test is passing! You can check by yourself. Please [go on this commit, which contains all of this code to this part and check it by yourself](https://github.com/sierisimo/publications/commit/929e3bb3dfff455fed5dccb51a2f716e5e43ebeb).
+
+At this point you probably are thinking I'm crazy, but I'm not. This is a fair thing happening on TDD, you created a test, you create code to pass the test, let's move on. No one said the implementation is right but the test is passing. We need to test deeper
+
+```kotlin
+@Test
+fun `jump function won't move to instruction X when registries are different`() {
+    //Setup
+    instructionSet.current = 3
+
+    zero(registry, 5)
+    zero(registry, 10)
+    increment(registry, 10)
+
+    jump(registry, 5, 10, instructionSet, 10)
+
+    //Assertions
+    assertNotEquals(10, instructionSet.current)
+    //If the jump function did not matched it should go in the next instruction
+    assertEquals(4, instructionSet.current)
+}
+```
+
+Now if we `check` the project we will get a failing test! This means our other test was right in the way we wrote it but the implementation of the function is actually wrong. Let's fix that.
+We need to validate that values at registers are equal, only then we update the postion on the `instructionSet`. Otherwise we should just continue execution on the next instruction. Adding the minimum to pass the test our test will pass again:
```

Comment:
```suggestion
We need to validate that values at registers are equal, only then we will update the position on the `instructionSet`. Otherwise we should just continue execution on the next instruction. Adding the minimum to pass the test our test will pass again:
```
Typo-po-po-po!

Reply:
Always on a hurry makes me do more typos